server/datastore/mysql/software_installers.go

```diff
@@ -566,3 +567,22 @@ ON DUPLICATE KEY UPDATE
 		return nil
 	})
 }
+
+func (ds *Datastore) HasSelfServiceSoftwareInstallers(ctx context.Context, platform string, teamID *uint) (bool, error) {
+	if fleet.IsLinux(platform) {
+		platform = "linux"
+	}
+	stmt := `SELECT 1 FROM software_installers WHERE self_service = 1 AND platform = ? AND (global_or_team_id = 0`
```

Comment:
That is correct for now. But we have a priority request for customer-preston to add software to "No Team": https://github.com/fleetdm/fleet/issues/19550
So, I'm future proofing

Reply:
Ok, would you please add some comments (or maybe better some specific test cases) to document this? As you've written it, both (1) hosts assigned to no team and (2) hosts assigned to a team without self-service software should never see self-service currently because there should never be any results that match the `global_or_team_id = 0` condition as long as we prevent users from uploading installers without a team id. 

I'm not sure we can safely assume that what emerges from the design process will work in this exact manner though. For example, it may be decided that hosts assigned to a team should only see self-service if that team has self-service software (i.e. "no team" software is only available to "no team" hosts).

Reply:
Ok, I simplified this to only look at `global_or_team_id`, which matches the INSERT statement in `MatchOrCreateSoftwareInstaller`